kore/test/Test/Kore/Internal/TermLike.hs

```diff
@@ -515,3 +536,61 @@ test_renaming =
                 updatesFreeVariables renamed
                 doesNotCapture (inject Mock.setY) renamed
             ]
+
+test_toSyntaxPattern :: TestTree
+test_toSyntaxPattern = Hedgehog.testProperty "convert a valid pattern to a Syntax.Pattern and back" . Hedgehog.property $ do
+    trmLike <- forAll (runKoreGen Mock.generatorSetup ConsistentKore.termLikeGen)
+    let patt = fmap (const Attribute.Null) (from trmLike :: Pattern.Pattern VariableName (TermAttributes VariableName))
+
+    case PatternVerifier.runPatternVerifier context $
+        PatternVerifier.verifyStandalonePattern Nothing patt of
+        Left Kore.Error.Error{errorError} ->
+            fail $
+                unparseToString patt <> "\n\n"
+                    <> show errorError
+        Right trmLike2 -> patt === fmap (const Attribute.Null) (from trmLike2 :: Pattern.Pattern VariableName (TermAttributes VariableName))
```

Comment:
This test isn't enitrely satisfactory. It only tests that going from `TermLike -> Syntax.Pattern -> TermLike -> Syntax.Pattern` is the same as `TermLike -> Syntax.Pattern`.
The difficulty with showing `TermLike -> Syntax.Pattern -> TermLike` is equivalent to the original `TermLike` lies in the fact that InternalList/Map/Set do not get internalized until simplification. I have tried running the simplifier on both the original and round trip `TermLike` to see if they end up equivalent, but unfortunately, they still differ syntactically. Perhaps all we can do is say that semantically they are the same? However, I'm not sure how to encode this property. @JKTKops @ana-pantilie any ideas? If not, I will leave the test as is. It is at least a good sanity check.